File: src/cpp_pkg/src/parameter.cpp
#include "rclcpp/rclcpp.hpp"

using namespace std;

class MinimalParameterNode : public rclcpp::Node {
  public:
    // 构造函数，初始化节点并声明参数
    MinimalParameterNode() : Node("minimal_parameter_node") {
        this->declare_parameter("custom_parameter", "Hello, World!");
        timer_ = this->create_wall_timer(1000ms, bind(&MinimalParameterNode::callback, this));
    }

  private:
    rclcpp::TimerBase::SharedPtr timer_;

    // 定时器回调函数，读取并打印参数值
    void callback() {
        string param = this->get_parameter("custom_parameter").as_string();
        RCLCPP_INFO(this->get_logger(), "Custom parameter: %s", param.c_str());
    }
};

int main(int argc, char **argv) {
    // 初始化 ROS 2
    rclcpp::init(argc, argv);
    // 创建并运行节点
    rclcpp::spin(std::make_shared<MinimalParameterNode>());
    // 关闭 ROS 2
    rclcpp::shutdown();
    return 0;
}

File: src/py_pkg/py_pkg/parameter.py
import rclpy
from rclpy.node import Node


class MinimalParameterNode(Node):
    # 初始化节点并声明一个自定义参数
    def __init__(self):
        super().__init__("minimal_parameter_node")
        self.declare_parameter("custom_parameter", "Hello, World!")
        self.__timer = self.create_timer(1.0, self.__callback)

    # 定时器回调函数，用于获取并打印自定义参数的值
    def __callback(self):
        param = self.get_parameter("custom_parameter")
        self.get_logger().info(f"Custom parameter: {param.value}")


def main(args=None):
    # 初始化 ROS 2 客户端库
    rclpy.init(args=args)
    # 创建节点并开始处理事件
    rclpy.spin(MinimalParameterNode())
    # 关闭 ROS 2 客户端库
    rclpy.shutdown()


if __name__ == "__main__":
    main()
